### ISSUE TITLE
Make space after semi-colon optional

### ISSUE THREAD

**Issue by johnmay** [dan2097/opsin#235]
Some salt names use a semi-colon to indicate the salt. OPSIN supports this is but only as ``"; "``

Example name

``(2S,4R)-1-[(2S)-2-amino-3,3-dimethyl-butanoyl]-4-hydroxy-N-[[4-(4-methylthiazol-5-yl)phenyl]methyl]pyrrolidine-2-carboxamide;hydrochloride``

**Comment by dan2097** [comment]
`; ` is implemented in a really weird way (it's fudged as a space). I think it should be simple enough to support semicolon more like how space/hyphen are allowed as delimiters between words. I should have time to finish testing the fix later this week.

**Comment by johnmay** [comment]
No rush, I fudge it in the reaction extractor anyways :-). Just thought a useful addition/relaxation.

**Comment by dan2097** [comment]
> No rush, I fudge it in the reaction extractor anyways :-). Just thought a useful addition/relaxation.

Yeah I think it's definitely a good idea. My initial idea of treating this identically to space/hyphen was too simplistic as while say `benzene;toluene` is probably fine I don't think `ethyl;benzene` should be interpreted as `ethylbenzene`
